pkg/ccl/streamingccl/streamclient/random_stream_client.go

```diff
@@ -407,7 +407,9 @@ func (m *randomStreamClient) Subscribe(
 			if event.Type() == streamingccl.KVEvent {
 				// Use the originally generated KeyValue copy as the KeyValue inside the event might
 				// get modified by ingestion processor's tenant rekeyer.
-				event = streamingccl.MakeKVEvent(*keyValCopy)
+				// 'keyValCopy' will only be set when it is a KV event. Copying the 'keyValCopy' again
+				// to prevent the event being modified by interceptor again.
+				event = streamingccl.MakeKVEvent(*copyKeyVal(keyValCopy))
```

Comment:
perhaps are apis are not good? I don't understand why we need a copy of a copy...